docs/manual/registration/objects-api.rst

```diff
@@ -0,0 +1,174 @@
+behandelproces.
+
+.. note:: De functioneel beheerder dient een aantal
+   :ref:`koppelingen <configuration_registration_objects>` in te stellen om deze plugin
+   te kunnen gebruiken. Let ook op de
+   :ref:`adviezen (Engels) <configuration_registration_objects_objecttype_tips>` voor
+   objecttypedefinities.
+
+Er zijn twee mogelijke manieren om de Objecten API in te zetten:
+
+* :ref:`manual_registration_objects_api_variables` (aangeraden)
+* :ref:`manual_registration_objects_api_templates` (verouderd)
+
+Deze worden hieronder toegelicht.
+
+.. _manual_registration_objects_api_variables:
+
+Variabelekoppelingen
+====================
+
+.. versionadded:: 2.6.0
+
+    Deze methode is nieuw sinds Open Formulieren 2.6 en biedt een robuustere manier van
+    werken. We raden sterk aan om enkel deze methode te gebruiken.
+
+De variabelekoppelingenconfiguratie laat toe om aan strikte objecttypedefinities te
+voldoen. Deze worden typisch ingericht als "contract" tussen het formulier en de
+afhandelapplicatie. Hiermee is de integratie een stuk robuuster en kan je makkelijker
+het formulier aanpassen/herorganiseren zonder dat de afhandelapplicatie hierdoor
+aanpassingen moet maken.
+
+**Mechanisme**
+
+* Elk formulierveld leidt tot een formuliervariabele
+* Na selectie van het relevante objecttype "kent" Open Formulieren het contract
+* Formuliervariabelen koppel je vervolgens aan bestemmingen die beschreven staan in het
+  contract
+* Tijdens het registeren voldoend de gegevens automatisch aan het beschreven contract
+  en heeft de afhandelapplicatie inhoudelijk begrip van de gegevens
+
+Voorbeeld
+---------
+
+**Objecttype**
+
+Voor dit voorbeeld gaan we uit van een objecttype waarin drie gegevens vastgelegd worden:
+
+* betaling voltooid (ja/nee)
+* groep "data"
+
+    * voornaam (tekst)
+    * leeftijd (getal, aantal jaren)
+
+.. note:: Technisch worden deze gegevens met JSON Schema gemodelleerd, en dat ziet er
+   dan zo uit:
+
+   .. code-block:: json
+
+     {
+       "$schema": "http://json-schema.org/draft-07/schema#",
+       "type": "object",
+       "properties": {
+         "payment_completed": {"type": "boolean"},
+         "data": {
+           "type": "object",
+           "properties": {
+             "firstName": {"type": "string"},
+             "age": {"type": "integer"}
+           }
+         }
+       }
+     }
+
+**Formulier**
+
+We gaan uit van een formulier met één stap. In deze stap zijn twee velden gedefinieerd:
+
+* Een tekstveld met de sleutel ``voornaam``
+* Een nummerveld met de sleutel ``leeftijd``
+
+Merk op dat de velden Nederlandse sleutels hebben en het objecttype Engelse sleutels
+gebruikt (voornaam vs. firstName, leeftijd vs. age).
+
+**Registratiemethode selecteren**
+
+Op de **Registratie** tab selecteer je als registratiemethod "Objecten API registratie".
+
+Vervolgens kies je de tab **Variabelekoppelingen**, waar je het objecttype kan selecteren
```

Comment:
Maybe here we should warn the user that if they are editing an existing form and they switch tab, the old configuration will be lost?

Reply:
They get the pop-up in the application, and I think we should aim the documentation on "new" forms rather than migrating existing forms.